volatility3/framework/symbols/windows/extensions/callbacks.py

```diff
@@ -24,27 +25,54 @@ def is_valid(self) -> bool:
                 and self.Entry.Blink.is_readable()
                 and self.DeviceObject.is_readable()
             ):
+                vollog.debug(
+                    f"Callback obj 0x{self.vol.offset:x} invalid due to unreadable structure members"
+                )
                 return False
 
+        except exceptions.InvalidAddressException:
+            vollog.debug(
+                f"callback obj 0x{self.vol.offset:x} invalid due to invalid address access"
+            )
+            return False
+
+        return True
+
+    def is_parseable(self, type_map: Dict[int, str]) -> bool:
+        """
+        Determines whether or not this `_SHUTDOWN_PACKET` callback can be reliably parsed.
+        Requires a `type_map` that maps NT executive object type indices to string representations.
+        This type map can be acquired via the `handles.Handles.get_type_map` classmethod.
+        """
+        if not self.is_valid():
+            return False
+
+        try:
+
             device = self.DeviceObject
             if not device or not (device.DriverObject.DriverStart % 0x1000 == 0):
                 vollog.debug(
                     f"callback obj 0x{self.vol.offset:x} invalid due to invalid device object"
                 )
                 return False
 
+            header = device.get_object_header()
+            object_type = header.get_object_type(type_map)
+            is_valid = object_type == "Device"
+            if not is_valid:
+                vollog.debug(
+                    f"Callback obj 0x{self.vol.offset:x} invalid due to invalid device type: wanted 'Device', found '{object_type}'"
+                )
+            return is_valid
         except exceptions.InvalidAddressException:
             vollog.debug(
                 f"callback obj 0x{self.vol.offset:x} invalid due to invalid address access"
             )
             return False
-
-        try:
-            header = device.get_object_header()
```

Comment:
Looks like this test got remove from `is_valid`, which *could* return different results, but I'm kinda willing to overlook it since I imagine it would only affect a very small number of results.